src/codemodder/codemods/regex_transformer.py

```diff
@@ -0,0 +1,60 @@
+import re
+from typing import Pattern
+
+from codemodder.codemods.base_transformer import BaseTransformerPipeline
+from codemodder.codetf import Change, ChangeSet
+from codemodder.context import CodemodExecutionContext
+from codemodder.diff import create_diff
+from codemodder.file_context import FileContext
+from codemodder.logging import logger
+from codemodder.result import Result
+
+
+class RegexTransformerPipeline(BaseTransformerPipeline):
+
+    def __init__(self, pattern: Pattern, replacement: str, change_description: str):
+        super().__init__()
+        self.pattern = pattern
+        self.replacement = replacement
+        self.change_description = change_description
+
+    def apply(
+        self,
+        context: CodemodExecutionContext,
+        file_context: FileContext,
+        results: list[Result] | None,
+    ) -> ChangeSet | None:
+
+        changes = []
+        updated_lines = []
+
+        with open(file_context.file_path, "r", encoding="utf-8") as f:
```

Comment:
`file_context.file_path` is a `Path` so you can use convenience methods like `read_text`.

Reply:
in this case we do want readlines because we need a line by line check so we can create the lineno change. below I can change